### DISCUSSION TITLE
API changes announcements

### DISCUSSION THREAD

**Discussion by stassius** [Mikubill/sd-webui-controlnet#1088] [Ideas]
Can we get announcements whenever API changes? There are plenty of people who develop plugins for ControlNet and every update we have to check the source code in order to understand changes. 

For example the last update gave us a new parameter called control_mode. But also there is still the guess_mode parameter in API. Will it be deprecated? If so, when? 

It would be really good if after each major update we got a short message describing what's going on in the API.

**Comment by ljleb** [comment]
The /controlnet/txt2img and /controlnet/img2img routes are deprecated. The api fields have been updated recently in the wiki. Please use the wiki for guidance about usage.

the api parameters displayed in /docs are deprecated, we should remove these routes eventually. How about we use discussions here for important changes?

**Comment by stassius** [comment]
Hi! This discussion is closed but I don't see any discussion posts regarding the API changes. 
Can you please create a discussion topic with all the API changes announcements? Or even a Discussion section about it?
For example, in the last update you have deprecated the Guess mode in the API. Were there any announcements about it? Where can I read them?

There are developers who rely on such information and just throwing warnings in the console is not a good way to inform people about changes.

**Reply by ljleb** [reply]
You can find a discussion of the removal of deprecated routes enclosing this current discussion here: https://github.com/Mikubill/sd-webui-controlnet/discussions/1132

tl;dr: the routes are now removed. Let us know in #1132 if you are against the changes and why, we will revert for important use cases that cannot update for understandable reasons.

**Comment by stassius** [comment]
I'm not talking about routes, I'm talking about deprecating individual parameters like Guess Mode or adding new parameters to API. 

Whenever there is a change like this in API which leads to changes in the external code, it would be great to have a short message about it in a dedicated discussion section. 

If you're pushing a commit with changes in API, explain, what did you change, which parameters were added with which ranges, what was removed, that's all.

**Reply by ljleb** [reply]
I agree. I think removing `guess_mode` completely is not a good change, as it is not necessary and eventually breaks existing code. If we really want to, we can make this information clear in the code and through the terminal that `control_mode` should be preferred over it. It may slightly increase complexity in the code, but I believe this is necessary to keep compatibility with other extensions and code.

For reference, the PR that removed guess mode completely is this one: https://github.com/Mikubill/sd-webui-controlnet/pull/1156

**Reply by dbokser** [reply]
what are the input options that are expected for control_mode? Are there updated API docs anywhere since the webui docs are outdated?

**Reply by stassius** [reply]
> what are the input options that are expected for control_mode? Are there updated API docs anywhere since the webui docs are outdated?

AFAIK, it's an integer value. 
0 for Balanced, 1 for Prompt priority and 2 for ControlNet priority.

**Reply by ljleb** [reply]
Check the wiki here, we keep it up to date relatively often.
